db/push.go

```diff
@@ -0,0 +1,96 @@
+package db
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	nomsjson "roci.dev/diff-server/util/noms/json"
+)
+
+// These request/response types might want to live in a different location, eg 
```

Comment:
I guess it makes sense to move things close to where they are used rather than a giant `types` pile.

Reply:
i think they should probably go in diffserver which is where we seem to be putting common code